src/config.rs

```diff
@@ -621,4 +620,284 @@ Message w/out In-Reply-To
 
         Ok(())
     }
+
+    // TODO refactoring: Should this really be placed here? But I wouldn't know a better place, either
```

Comment:
I originally put the AEAP tests here in order to put them next to `test_change_primary_self_addr()`. `test_change_primary_self_addr()`, in turn, fits here nicely because `set_primary_self_addr()` is in `config.rs`.

However, all the `test_aeap*` tests have nothing to do with AEAP, so they don't actually fit here. I could but them into `dc_receive_imf.rs`, `mimeparser.rs`, `peerstate.rs` or a new file `aeap.rs`.

Reply:
We can't make them true integration tests, living outside of `src`, since they need access to some non-pub API like `sql`. But what we can do is create a `tests` folder inside `src`. (also see https://matklad.github.io/2021/02/27/delete-cargo-integration-tests.html, and https://github.com/rust-lang/rust-analyzer/tree/master/crates/hir-ty/src/tests for a crate that does that)

Reply:
I did so in 7c0b2b4e764d5815e2372a66b1f54e9016fdc7ef.

It would generally be nicer to have an actual integration test, since it would be a separate crate, probably resulting in faster compilation. I just played around with this a bit; the main problem I found is that there seems to be no way to enable a feature in the `deltachat` crate when compiling it for our integration test. But we would need this in order to make some things public (like the `internals` feature does) and make sure that `test_utils.rs` is compiled.